flake.nix

```diff
@@ -1,45 +1,56 @@
 {
   description = "Build NPM packages in Nix and lightweight NPM registry";
+
   inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
   inputs.flake-utils.url = "github:numtide/flake-utils";
 
   outputs = { self, nixpkgs, flake-utils }:
-    let
-      internal_overlay = final: prev: {
-        napalm = import ./. {
-          pkgs = final;
-        };
-      };
-    in
     flake-utils.lib.eachDefaultSystem
-      (
-        system:
+      (system:
         let
-          pkgs = import nixpkgs { inherit system; overlays = [ internal_overlay ]; };
+          napalm = import ./. {
+            pkgs = nixpkgs.legacyPackages.${system};
+          };
         in
         {
-          packages = {
-            inherit (pkgs.napalm)
-              hello-world hello-world-deps netlify-cli deckdeckgo-starter
-              bitwarden-cli napalm-registry
-              ;
+          packages = (nixpkgs.lib.filterAttrs (_: nixpkgs.lib.isDerivation) napalm) // {
```

Comment:
I think I prefer the explicit package lists – that way it is clear what is exported and devshell is not included among packages.